test/python/algorithms/test_optimizer_de.py

```diff
@@ -0,0 +1,64 @@
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test of differential evolution optimizer."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from scipy.optimize import rosen
+import numpy as np
+
+from qiskit import BasicAer
+from qiskit.algorithms import VQE
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit.opflow import PauliSumOp
+from qiskit.algorithms.optimizers import DE
+
+
+class TestOptimizerDE(QiskitAlgorithmsTestCase):
+    """ Test Differential Evolution (DE) optimizer"""
+
+    def setUp(self):
+        super().setUp()
+        self.seed = 50
+        algorithm_globals.random_seed = self.seed
+        self.qubit_op = PauliSumOp.from_list([
+            ("II", -1.052373245772859),
+            ("IZ", 0.39793742484318045),
+            ("ZI", -0.39793742484318045),
+            ("ZZ", -0.01128010425623538),
+            ("XX", 0.18093119978423156),
+        ])
+
+    def _optimize(self, optimizer):
```

Comment:
This looks like it's missing something. Should the `optimizer` here be set?

Reply:
Thanks for catching this @Cryoris. I had put this here because the scipy differential evolution optimizer needs `variable_bounds` to be specified. I will refactor and migrate this to `test/python/algorithms/test_optimizers.py` as requested below

Reply:
Updated